lib/parallel_tests/gherkin/runner.rb

```diff
@@ -67,11 +70,25 @@ def summarize_results(results)
         end
 
         def cucumber_opts(given)
-          if given =~ (/--profile/) || given =~ (/(^|\s)-p /)
-            given
-          else
-            [given, profile_from_config].compact.join(" ")
+          initial =
+            if given =~ (/--profile/) || given =~ (/(^|\s)-p /)
+              given
+            else
+              [given, profile_from_config].compact.join(" ")
+            end
+
+          opts_as_individuals = initial.scan(/\S+\s\S+/)
+          desired_output = ['', '']
+
+          opts_as_individuals.each do |opt|
```

Comment:
Here we iterate over each of them, and re-assemble them back how they were, with the one exception, --retry goes into a separate array. So the above example would be turned into an array of the 4 items in position 0 and --retry 3 in position 1

Reply:
Note to self - This might be more desirable

```
2.7.3 :027 > array.each_with_index do |item, index|
2.7.3 :028 >   if item.match?(/--retry \d+/)
2.7.3 :029 >     @desired = array.delete_at(index)
2.7.3 :030 >   end
2.7.3 :031 > end
 => ["-p foo", "-p bar", "-t @foo", "-t @bar"] 
2.7.3 :032 > @desired
 => "--retry 3" 
```